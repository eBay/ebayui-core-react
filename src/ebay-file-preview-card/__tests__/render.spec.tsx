import React from 'react'
import { render } from '@testing-library/react'
import { EbayFilePreviewCard } from '../'

describe('<EbayFilePreviewCard>', () => {
    it('renders with status uploading', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                status="uploading"
                a11yCancelUploadText="Cancel upload"
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders as span element', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                as='span'
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.jpg',
                    type: 'image',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
                footerTitle="file-name.csv"
                footerSubtitle="English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic"
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders with image, delete button and footer', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.jpg',
                    type: 'image',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
                footerTitle="file-name.csv"
                footerSubtitle="English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic"
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders with video', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.jpg',
                    type: 'image/jpeg',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
                menuActions={[
                    {
                        event: 'edit',
                        label: 'Edit'
                    },
                    {
                        event: 'download',
                        label: 'Download'
                    }
                ]}
            />
        )
        const snapshot = asFragment()
        const fakeMenuButtonElement = snapshot.querySelectorAll(
            'button.menu-button__button'
        )
        fakeMenuButtonElement.forEach((button) => {
            button.setAttribute('aria-controls', '1234')
        })
        expect(snapshot).toMatchSnapshot()
    })
    it('renders with document', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.jpg',
                    type: 'image/jpeg',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
                menuActions={[
                    {
                        event: 'edit',
                        label: 'Edit'
                    },
                    {
                        event: 'download',
                        label: 'Download'
                    }
                ]}
            />
        )
        const snapshot = asFragment()
        const fakeMenuButtonElement = snapshot.querySelectorAll(
            'button.menu-button__button'
        )
        fakeMenuButtonElement.forEach((button) => {
            button.setAttribute('aria-controls', '1234')
        })
        expect(snapshot).toMatchSnapshot()
    })
    it('renders with multi action button', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.jpg',
                    type: 'image/jpeg',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
                menuActions={[
                    {
                        event: 'edit',
                        label: 'Edit'
                    },
                    {
                        event: 'download',
                        label: 'Download'
                    }
                ]}
            />
        )
        const snapshot = asFragment()
        const fakeMenuButtonElement = snapshot.querySelectorAll(
            'button.menu-button__button'
        )
        fakeMenuButtonElement.forEach((button) => {
            button.setAttribute('aria-controls', '1234')
        })
        expect(snapshot).toMatchSnapshot()
    })
    it('renders with see more button', () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                a11ySeeMoreText="See more"
                seeMore={15}
                file={{
                    name: 'file-name.jpg',
                    type: 'image/jpeg',
                    src: 'https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg'
                }}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it('should render with video', async () => {
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={{
                    name: 'file-name.mov',
                    type: 'video/quicktime',
                    src: 'https://storage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4'
                }}
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
    it('should render with mock file', async () => {
        const mockFile = new File(['file content'], 'example.txt', {
            type: 'text/plain'
        })
        const { asFragment } = render(
            <EbayFilePreviewCard
                a11yCancelUploadText="Cancel upload"
                deleteText="Delete"
                file={mockFile}
                footerTitle="file-name.csv"
                footerSubtitle="English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic"
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
