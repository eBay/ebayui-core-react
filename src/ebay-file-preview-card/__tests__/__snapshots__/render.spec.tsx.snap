// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayFilePreviewCard> renders as span element 1`] = `
<DocumentFragment>
  <span
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <button
        aria-label="Delete"
        class="file-preview-card__action icon-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-delete-16"
          />
        </svg>
      </button>
    </div>
    <div
      class="file-preview-card__footer"
    >
      <span>
        file-name.csv
      </span>
      <span>
        English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic
      </span>
    </div>
  </span>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with document 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <span
        class="menu-button file-preview-card__action"
      >
        <button
          aria-controls="1234"
          aria-expanded="false"
          aria-haspopup="true"
          class="menu-button__button icon-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-overflow-vertical-16"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with image, delete button and footer 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <button
        aria-label="Delete"
        class="file-preview-card__action icon-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-delete-16"
          />
        </svg>
      </button>
    </div>
    <div
      class="file-preview-card__footer"
    >
      <span>
        file-name.csv
      </span>
      <span>
        English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with multi action button 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <span
        class="menu-button file-preview-card__action"
      >
        <button
          aria-controls="1234"
          aria-expanded="false"
          aria-haspopup="true"
          class="menu-button__button icon-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-overflow-vertical-16"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with see more button 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset file-preview-card__asset--fade"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <button
        aria-label="See more"
        class="file-preview-card__see-more"
        type="button"
      >
        <span>
          +15
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with status uploading 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <span
        aria-label="Busy"
        class="progress-spinner file-preview-card__asset"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--24"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner-24"
          />
        </svg>
      </span>
      <button
        aria-label="Cancel upload"
        class="file-preview-card__action icon-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-close-16"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> renders with video 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <img
        alt="file-name.jpg"
        class="file-preview-card__asset"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
      />
      <span
        class="menu-button file-preview-card__action"
      >
        <button
          aria-controls="1234"
          aria-expanded="false"
          aria-haspopup="true"
          class="menu-button__button icon-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-overflow-vertical-16"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> should render with mock file 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <svg
        aria-hidden="true"
        class="file-preview-card__asset icon icon--24"
        focusable="false"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-file-24"
        />
      </svg>
      <button
        aria-label="Delete"
        class="file-preview-card__action icon-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-delete-16"
          />
        </svg>
      </button>
      <div
        class="file-preview-card__info"
      >
        TXT
      </div>
    </div>
    <div
      class="file-preview-card__footer"
    >
      <span>
        file-name.csv
      </span>
      <span>
        English, German, Spanish, French, Polish, Dutch, Italian, Japanese, Portuguese, Arabic
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayFilePreviewCard> should render with video 1`] = `
<DocumentFragment>
  <div
    class="file-preview-card"
  >
    <div
      class="file-preview-card__body"
    >
      <video
        class="file-preview-card__asset"
        src="https://storage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4"
      />
      <button
        aria-label="Delete"
        class="file-preview-card__action icon-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-delete-16"
          />
        </svg>
      </button>
      <div
        class="file-preview-card__info"
      >
        <svg
          aria-hidden="true"
          class="file-preview-card__video-icon icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-play-16"
          />
        </svg>
        MOV
      </div>
    </div>
  </div>
</DocumentFragment>
`;
